nifi-rdf4j-repository-put-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/processors/LdesTriplestoreMaterialisation.java

```diff
@@ -0,0 +1,183 @@
+
+import org.apache.nifi.processor.util.StandardValidators;
+
+@Tags({ "ldes-triplestore, vsds" })
+@CapabilityDescription("Materialises LDES events in a triplestore (SPARQL query)")
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+public class LdesTriplestoreMaterialisation extends AbstractProcessor {
+	private RepositoryManager repositoryManager;
+
+	static final Relationship REL_SUCCESS = new Relationship.Builder()
+			.name("success")
+			.description("A FlowFile is routed to this relationship after the database is successfully updated")
+			.build();
+	static final Relationship REL_FAILURE = new Relationship.Builder()
+			.name("failure")
+			.description("A FlowFile is routed to this relationship if the database cannot be updated and retrying the operation will also fail, "
+					+ "such as an invalid query or an integrity constraint violation")
+			.build();
+
+	static final PropertyDescriptor SPARQL_HOST = new PropertyDescriptor.Builder()
+			.name("SPARQL host")
+			.description("The hostname and port of the SPARQL server.")
+			.defaultValue("http://graphdb:7200")
+			.required(true)
+			.addValidator(StandardValidators.URL_VALIDATOR)
+			.build();
+
+	static final PropertyDescriptor REPOSITORY_ID = new PropertyDescriptor.Builder()
+			.name("Repository ID")
+			.description("The repository to connect to.")
+			.required(true)
+			.addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+			.build();
+
+	static final PropertyDescriptor NAMED_GRAPH = new PropertyDescriptor.Builder()
+			.name("Named graph")
+			.description("If set, the named graph the triples will be written to.")
+			.required(false)
+			.addValidator(StandardValidators.URI_VALIDATOR)
+			.build();
+
+
+	@OnScheduled
+	public void onScheduled(final ProcessContext context) {
+		this.repositoryManager = new RemoteRepositoryManager(context.getProperty(SPARQL_HOST).getValue());
+	}
+
+	@Override
+	public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+		// @todo Find out the ideal number of FlowFiles to process in one transaction.
+		final List<FlowFile> flowFiles = session.get(50);
+
+		Repository repository = repositoryManager.getRepository(context.getProperty(REPOSITORY_ID).getValue());
+		if (flowFiles.isEmpty()) {
+			return;
+		}
+		final AtomicBoolean committed = new AtomicBoolean(false);
+		try (RepositoryConnection dbConnection = repository.getConnection()) {
+			// As we are bulk-loading, set isolation level to none for improved performance.
+			dbConnection.setIsolationLevel(IsolationLevels.NONE);
+			dbConnection.begin();
+
+			for (FlowFile flowFile : flowFiles) {
+				session.read(flowFile, new InputStreamCallback() {
+					@Override
+					public void process(InputStream in) throws IOException {
+						String LDESMemberRDF = IOUtils.toString(in);
+						InputStream targetStream = IOUtils.toInputStream(LDESMemberRDF);
+						Model updateModel = Rio.parse(targetStream, "", RDFFormat.NQUADS);
```

Comment:
Can this not be replaced by the following:
```suggestion
Model updateModel = Rio.parse(in, "", RDFFormat.NQUADS);
```